tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Services/FailureAnalysis/MavenBrokenPipeFailureClassifier.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Azure.Sdk.Tools.PipelineWitness.Entities.AzurePipelines;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.PipelineWitness.Services.FailureAnalysis
+{
+    public class MavenBrokenPipeFailureClassifier : IFailureClassifier
+    {
+        public MavenBrokenPipeFailureClassifier(VssConnection vssConnection)
+        {
+            this.vssConnection = vssConnection;
+            buildClient = vssConnection.GetClient<BuildHttpClient>();
+        }
+
+        private VssConnection vssConnection;
+        private BuildHttpClient buildClient;
+
+        public async Task ClassifyAsync(FailureAnalyzerContext context)
+        {
+            var failedTasks = from r in context.Timeline.Records
+                              where r.Result == TaskResult.Failed
+                              where r.RecordType == "Task"
+                              where r.Task.Name == "Maven"
+                              select r;
+
+            foreach (var failedTask in failedTasks)
+            {
+                var lines = await buildClient.GetBuildLogLinesAsync(
+                    context.Build.Project.Id,
+                    context.Build.Id,
+                    failedTask.Log.Id
+                    );
+
+                if (lines.Any(line => line.Contains("Connection reset")))
```

Comment:
Is this the only download failure message we have seen so far?

Reply:
Its the predominant one. There is one more I think but I couldn't find an example, when I spot one I'll add it to the search condition.